### DISCUSSION TITLE
Tiddloid 2.0 early access

### DISCUSSION THREAD

**Discussion by donmor** [TiddlyWiki/TiddlyWiki5#5764] [General]
Finally I got ready to merge all good things of Tiddloid Lite into Tiddloid, which will implement new APIs and work well on new devices.

Here's a link to the pre-release page: https://github.com/donmor/Tiddloid/releases/tag/2.0.0a

and a latest debug apk ahead of the pre-release with several bug fixed (change extension to apk): 
[app-debug.zip](https://github.com/Jermolene/TiddlyWiki5/files/6603673/app-debug.zip)

**Comment by donmor** [comment]
And here's today's build:
[app-debug.zip](https://github.com/Jermolene/TiddlyWiki5/files/6612899/app-debug.zip)
Fixed a bunch of bugs.

**Comment by donmor** [comment]
A more complete pre-release came out. Check it [here](https://github.com/donmor/Tiddloid/releases/tag/2.0.0a2).

**Reply by AnthonyMuscio** [reply]
Thanks for maintaining this!

Tony

**Comment by donmor** [comment]
Finally here comes the Release 2.0.0! Here's the link: https://github.com/donmor/Tiddloid/releases/latest
#### Update details:
- Refactored file system implementation and now it mainly use Storage Access Framework (lecacy file:// mode still initially supported).
- Now supports SAF folder mode by caching the selected directory (Requiring an index.html or index.htm in the folder, which will be automatically created if not exists).
- Backup system that supports SAF mode (in external files dir) and SAF folder mode.
- Refactored fork system to clone function.
  - Dropped web search mode
  - added support for receiving shared html file/text containing a url
  - added support for opening a http(s) uri/html file (if using a legacy file manager that produces file:// uri, a legacy wiki will be added)
  - in addition, added support for saving http(s):// link to the wiki list as a shortcut, which can be cloned to a file anytime you want.
- added a save button to the toolbar. You can save your work even if you hid your save button in wiki control panel.
- Fixed a bug causing malfunction of wiki detection with TWC.
- Fixed a bug causing mistaken permission revocation when having a page opened and perform a missing wiki auto-remove.
- Inspired by [this issue](https://github.com/donmor/TiddloidLite/issues/11), Now the modified data will be dumped into `INTERNAL/Android/data/top.donmor.tiddloid/files` if failed to save
- Wiki page's toolbar will no longer hide itself under landscape mode UNLESS configured using the plugin

**Reply by hubertgk** [reply]
Great work! Thank you for developing / maintaining the Tiddloid apps.

**Comment by linonetwo** [comment]
Do you think it is possible to support nodejs wiki's folder structure?

**Reply by donmor** [reply]
There's something called Termux...
Don't bother DIY if there's a better implementation XD
